heron/stmgr/tests/cpp/server/dummy_ckptmgr_client.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <iostream>
+#include <limits>
+#include <set>
+#include <string>
+#include "proto/messages.h"
+#include "basics/basics.h"
+#include "errors/errors.h"
+#include "threads/threads.h"
+#include "network/network.h"
+#include "server/dummy_ckptmgr_client.h"
+
+DummyCkptMgrClient::DummyCkptMgrClient(EventLoop* _eventLoop, const NetworkOptions& _options,
```

Comment:
Sailesh attempted this some time back and only did partial things(mostly in the networking world). The thing is that there is some extensive re jiggering needed to make our classes mock friendly. That's why we are using virtual methods for unit tests for now.